sycl/include/sycl/ext/oneapi/experimental/enqueue_functions.hpp

```diff
@@ -357,6 +288,75 @@ void nd_launch(queue Q, launch_config<nd_range<Dimensions>, Properties> Config,
   });
 }
 
+// Free function kernel enqueue functions
+template <auto *Func, typename... ArgsT>
+void single_task(queue Q, kernel_function_s<Func> KernelFunc, ArgsT &&...Args) {
+  (void)KernelFunc;
+  submit(Q, [&](handler &CGH) {
+    single_task(CGH, KernelFunc, std::forward<ArgsT>(Args)...);
+  });
+}
+
+template <auto *Func, typename... ArgsT>
+void single_task(handler &CGH, kernel_function_s<Func> KernelFunc,
+                 ArgsT &&...Args) {
+  (void)KernelFunc;
+  queue Q = CGH.getQueue();
+  sycl::kernel_bundle Bndl =
+      get_kernel_bundle<Func, sycl::bundle_state::executable>(Q.get_context());
```

Comment:
This creates and destroys two `std::shared_ptr`s for almost no reason. IMO, we should fix this `getQueue()` hack while we're in an ABI breaking window. Maybe by changing `handler_impl` to store a reference to the `sycl::queue` it was created with? `handler_impl::MQueueOrGraph` isn't used directly outside a few getters, so the change should be very simple.

Reply:
Can you elaborate a bit please?
What shared pointers are you referring to?